python/pyspark/testing/connectutils.py

```diff
@@ -167,6 +167,7 @@ def conf(cls):
         Override this in subclasses to supply a more specific conf
         """
         conf = SparkConf(loadDefaults=False)
+        conf.set("spark.sql.legacy.setCommandRejectsSparkCoreConfs", "false")
```

Comment:
why we need this change?

Reply:
Otherwise the line `conf.set("spark.connect.execute.reattachable.senderMaxStreamDuration", "1s")` will throw an exception

Reply:
Otherwise `test_basic_connect` fails with 

```

> ERROR [0.000s]: setUpClass (pyspark.sql.tests.connect.test_connect_basic.SparkConnectSessionWithOptionsTest)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/Users/m.zhang/spark/python/pyspark/testing/connectutils.py", line 187, in setUpClass
>     PySparkSession.builder.config(conf=cls.conf())
>   File "/Users/m.zhang/spark/python/pyspark/sql/session.py", line 480, in getOrCreate
>     return RemoteSparkSession.builder.config(map=opts).getOrCreate()
>   File "/Users/m.zhang/spark/python/pyspark/sql/connect/session.py", line 217, in getOrCreate
>     session = self.create()
>   File "/Users/m.zhang/spark/python/pyspark/sql/connect/session.py", line 208, in create
>     self._apply_options(session)
>   File "/Users/m.zhang/spark/python/pyspark/sql/connect/session.py", line 182, in _apply_options
>     session.conf.set(k, v)
>   File "/Users/m.zhang/spark/python/pyspark/sql/connect/conf.py", line 41, in set
>     result = self._client.config(operation)
>   File "/Users/m.zhang/spark/python/pyspark/sql/connect/client/core.py", line 1347, in config
>     self._handle_error(error)
>   File "/Users/m.zhang/spark/python/pyspark/sql/connect/client/core.py", line 1477, in _handle_error
>     self._handle_rpc_error(error)
>   File "/Users/m.zhang/spark/python/pyspark/sql/connect/client/core.py", line 1513, in _handle_rpc_error
>     raise convert_exception(info, status.message) from None
> pyspark.errors.exceptions.connect.AnalysisException: [CANNOT_MODIFY_CONFIG] Cannot modify the value of the Spark config: "spark.connect.execute.reattachable.senderMaxStreamDuration".
> See also 'https://spark.apache.org/docs/latest/sql-migration-guide.html#ddl-statements'.

```

Reply:
ConfigEntry's are added when they are created. [maxStreamDuration configuration option creation.](https://github.com/apache/spark/blob/b9c4fa4bc3129a352c443be04ef88f3ea71ae99d/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala#L112). 
[ConfigEntry constructor registers the config entry into the global config entry registry.](https://github.com/apache/spark/blob/b9c4fa4bc3129a352c443be04ef88f3ea71ae99d/core/src/main/scala/org/apache/spark/internal/config/ConfigEntry.scala#L87)

[RuntimeConf configuration checking](https://github.com/apache/spark/blob/b9c4fa4bc3129a352c443be04ef88f3ea71ae99d/sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala#L160) checks [spark.sql.legacy.setCommandRejectsSparkCoreConfs which defaults to true](https://github.com/apache/spark/blob/b9c4fa4bc3129a352c443be04ef88f3ea71ae99d/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L3672). Note that since `spark.sql.legacy.setCommandRejectsSparkCoreConfs` is not a SQLConf, it is not found in the SQLConf registry so `requireNonStatic` throws.

I'm using `spark.sql.legacy.setCommandRejectsSparkCoreConfs` to set non-sql configurations to get the test to run.